src/core/public/chrome/ui/header/header.tsx

```diff
@@ -17,187 +17,45 @@
-      defaultMessage: '{recentlyAccessedItemLinklabel}, type: {pageType}',
-      values: {
-        recentlyAccessedItemLinklabel: recentlyAccessed.label,
-        pageType: navLink.title,
-      },
-    });
-  }
-
-  return {
-    href,
-    label: truncateRecentItemLabel(recentlyAccessed.label),
-    title: titleAndAriaLabel,
-    'aria-label': titleAndAriaLabel,
-    euiIconType: navLink?.euiIconType,
-  };
-}
-
-function euiNavLink(
-  navLink: ChromeNavLink,
-  legacyMode: boolean,
-  currentAppId: string | undefined,
-  basePath: HttpStart['basePath'],
-  navigateToApp: CoreStart['application']['navigateToApp']
-) {
-  const {
-    legacy,
-    url,
-    active,
-    baseUrl,
-    id,
-    title,
-    disabled,
-    euiIconType,
-    icon,
-    category,
-    order,
-    tooltip,
-  } = navLink;
-  let href = navLink.baseUrl;
-
-  if (legacy) {
-    href = url && !active ? url : baseUrl;
-  }
-
-  return {
-    category,
-    key: id,
-    label: tooltip ?? title,
-    href, // Use href and onClick to support "open in new tab" and SPA navigation in the same link
-    onClick(event: MouseEvent) {
-      if (
-        !legacyMode && // ignore when in legacy mode
-        !legacy && // ignore links to legacy apps
-        !event.defaultPrevented && // onClick prevented default
-        event.button === 0 && // ignore everything but left clicks
-        !isModifiedEvent(event) // ignore clicks with modifier keys
-      ) {
-        event.preventDefault();
-        navigateToApp(navLink.id);
-      }
-    },
-    // Legacy apps use `active` property, NP apps should match the current app
-    isActive: active || currentAppId === id,
-    isDisabled: disabled,
-    iconType: euiIconType,
-    icon: !euiIconType && icon ? renderLinkIcon(basePath.prepend(`/${icon}`)) : undefined,
-    order,
-    'data-test-subj': 'navDrawerAppsMenuLink',
-  };
-}
-
-function renderLinkIcon(url: string) {
-  return <EuiImage size="s" alt="" aria-hidden={true} url={url} />;
-}
-
-function isModifiedEvent(event: MouseEvent) {
-  return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
-}
-
-function findClosestAnchor(element: HTMLElement): HTMLAnchorElement | void {
-  let current = element;
-  while (current) {
-    if (current.tagName === 'A') {
-      return current as HTMLAnchorElement;
-    }
-
-    if (!current.parentElement || current.parentElement === document.body) {
-      return undefined;
-    }
-
-    current = current.parentElement;
-  }
-}
-
-function truncateRecentItemLabel(label: string): string {
-  if (label.length > TRUNCATE_LIMIT) {
-    label = `${label.substring(0, TRUNCATE_AT)}…`;
-  }
-
-  return label;
-}
+import { HeaderBadge } from './header_badge';
+import { HeaderBreadcrumbs } from './header_breadcrumbs';
+import { HeaderHelpMenu } from './header_help_menu';
+import { HeaderNavControls } from './header_nav_controls';
+import { RecentLinks, NavLink, HeaderLogo, euiNavLink } from './header_bits';
```

Comment:
`header` already got it's own folder in `src/core/public/chrome/ui` and there are already some components at the same level. I think the `header_bits` subfolder is not necessary (or at least every subcomponents should be at the same place)